src/storage-controller/src/lib.rs

```diff
@@ -869,9 +893,6 @@ where
             }
         }
 
-        self.append_shard_mappings(new_collections.into_iter(), 1)
```

Comment:
AFAICT, this call was a no-op before because it would hit this short-circuit:

```
        let id = match self
            .introspection_ids
            .lock()
            .expect("poisoned")
            .get(&IntrospectionType::ShardMapping)
        {
            Some(id) => *id,
            _ => return,
        };
```

Reply:
Oh no, this was incorrect. It was a no-op for the first call, but not on the later ones